src/logic/calculate.js

```diff
@@ -0,0 +1,37 @@
+import operate from './operate';
+
+const calculate = (object, buttonNames) => {
+  const NewObj = { ...object };
+
+  let { total, next, operation } = NewObj;
+  if (buttonNames === 'AC') {
```

Comment:
- [ ] [REQUIRED] It will be nice if you implement a case if the button name is equal to ` . ` .